### DISCUSSION TITLE
Access baseURL from inside Admin for Widget Development

### DISCUSSION THREAD

**Discussion by xmdcode** [medusajs/medusa#5043] [Feature Requests]
Hello,

I have the following problem while helping to develop a custom plugin for medusajs (medusa-plugin-feeds).

The feeds are generated automatically with the <BACKEND_URL>/feed/xml.

Although i want to have access on the admin url to get the backend_url so i can show it on the widget along with the /feed/xml for easier access on the client so you can copy the whole link and share it to appropriate marketplace etc.

For now i can see which is the baseURL from const {client} = useMedusa(); but i can't access it since   Property 'config' is private and only accessible within class 'Client'. (baseURL variable is inside config).

Is there any plans in the future to be available for this purpose?

Thank you.

**Comment by pevey** [comment]
The admin should pull in MEDUSA_BACKEND_URL as well as any env vars prefixed with MEDUSA_ADMIN_.

Did you write BACKEND_URL above just for brevity?  Try using MEDUSA_BACKEND_URL.

**Comment by xmdcode** [comment]
I did that but it does't come up since it says "Cannot find name 'MEDUSA_BACKEND_URL'."

**Comment by pevey** [comment]
It's not elegant, but as a test, try duplicating the value and naming it MEDUSA_ADMIN_BACKEND_URL.  I'm wondering if only the variables prefixed with MEDUSA_ADMIN_ can be accessed in widgets.

**Reply by xmdcode** [reply]
I can't access it also "Cannot find name 'MEDUSA_ADMIN_BACKEND_URL'."

As i said i need to access this on the FRONTEND of admin inside `/src/admin/routes/feed` where frontend code lives.

**Reply by pevey** [reply]
The way I understood it (and you can find some info on this is you search Discord), srindom make it so that any env var prefixed with MEDUSA_ADMIN_ should be loaded by the admin app (using process.env) and made available to your widgets and routes in the same way that the MEDUSA_BACKEND_URL is loaded here: @medusajs/admin-ui/src/constants/medusa-backend-url.ts

**Reply by xmdcode** [reply]
You are right i was trying to console.log MEDUSA_BACKEND_URL or MEDUSA_ADMIN_BACKEND_URL without process.env. Too much code today ! Thank you!
